lib/signals.c

```diff
@@ -181,10 +181,8 @@ static void
 signal_handler(int sig)
 {
 	if (write(signal_pipe[1], &sig, sizeof(int)) != sizeof(int)) {
-		DBG("signal_pipe write error %s", strerror(errno));
 		assert(0);
-
-		log_message(LOG_INFO, "BUG - write to signal_pipe[1] error %s - please report", strerror(errno));
+		fprintf(stderr, "BUG - write to signal_pipe[1] error %s - please report\n", strerror(errno));
 	}
 }
 #endif
```

Comment:
Could you please post the output of `keepalived -v`. Since Linux 2.6.22 (and glibc 2.8) keepalived doesn't use signal handlers.

Reply:
My kernel is base on linux-3.10,but keepalived is still old version:
[root@localhost linux]# keepalived -v
Keepalived v1.3.5 (03/19,2017), git commit v1.3.5-6-g6fa32f2

Copyright(C) 2001-2017 Alexandre Cassen, <acassen@gmail.com>

Build options:  PIPE2 LIBNL3 RTA_ENCAP RTA_EXPIRES RTA_PREF FRA_OIFNAME FRA_TUN_ID RTAX_CC_ALGO RTAX_QUICKACK LIBIPTC LIBIPSET_DYNAMIC LVS LIBIPVS_NETLINK IPVS_64BIT_STATS VRRP VRRP_AUTH VRRP_VMAC SOCK_NONBLOCK SOCK_CLOEXEC FIB_ROUTING INET6_ADDR_GEN_MODE SNMP_V3_FOR_V2 SNMP SNMP_KEEPALIVED SNMP_CHECKER SNMP_RFC SNMP_RFCV2 SNMP_RFCV3 SO_MARK

[root@localhost linux]# uname -a
Linux localhost.localdomain 3.10.0-862.14.0.1.h43.eulerosv2r7.x86_64 #6 SMP Fri Dec 28 08:27:17 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux